synapse/app/_base.py

```diff
@@ -97,3 +97,58 @@ def run():
         daemon.start()
     else:
         run()
+
+
+def listen_tcp(logger, bind_addresses, port, factory, backlog=50):
```

Comment:
it'd probably be better to create a new `logger` in this file than to pass around the logger from another file.